### ISSUE TITLE
Shorthand for binding a class to itself

### ISSUE THREAD

**Issue by endel** [inversify/InversifyJS#311]
Hey there,

Would be nice to support binding a class to itself, such as:

``` typescript
kernel.bind(Something).inSingletonScope()

// Currently it's possible to achieve that using
kernel.bind(Something).to(Something).inSingletonScope()
```

What's your thoughts about this? Cheers!

**Comment by remojansen** [comment]
I think is a great idea :+1:  but for the API it would be easier to implement if we did something like:

``` ts
kernel.bind(Something).toSelf();
```

The `toSelf` method should return`BindingInWhenOnSyntax` so users can use the rest of the binding API:

``` ts
kernel.bind(Something).toSelf().inSingletonScope();
kernel.bind(Something).toSelf().whenTargetNamed("name");
kernel.bind(Something).toSelf().onActivation(() => { /* ... */ });
```

Also some users will be familiar with this because it is based on [Ninject](https://github.com/ninject/Ninject/wiki/Dependency-Injection-With-Ninject#skipping-the-type-binding-bit--implicit-self-binding-of-concrete-types).

**Comment by remojansen** [comment]
Closing this as it has been merged to master #312 :smiley:
